great_expectations/checkpoint/actions.py

```diff
@@ -165,6 +161,14 @@ class (for open source Great Expectations) or the GeCloudIdentifier (from Great
         """  # noqa: E501
         return NotImplementedError
 
+    # NOTE: To be promoted to 'run' after V1 development
+    def v1_run(self, checkpoint_result: CheckpointResult) -> None:
+        return self._v1_run(checkpoint_result=checkpoint_result)
+
+    # NOTE: To be promoted to '_run' after V1 development
+    def _v1_run(self, checkpoint_result: CheckpointResult):
+        raise NotImplementedError
```

Comment:
Can only be promoted once the legacy checkpoint is removed

Reply:
Having this just run on the total `CheckpointResult` - makes the API much much simpler and contains all the data we need (validation results, identifiers, suite and checkpoint information, etc)